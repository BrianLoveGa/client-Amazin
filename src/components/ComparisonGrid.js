import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});

function createData(name, power, autoPrograms, manPrograms, crushing) {
  return { name, power, autoPrograms, manPrograms, crushing };
}

const rows = [
  createData('Ninja Professional with Cups', 1100, 0, 4, 0),
  createData('Ninja Professional', 1000, 4, 4, 0),
  createData('Nutri Ninja DUo with Auto-iQ', 1200, 16.0, 24, 6.0),
  createData('Cupcake', 305, 3.7, 67, 4.3),
  createData('Gingerbread', 356, 16.0, 49, 3.9),
];

export default function SimpleTable() {
  const classes = useStyles();

  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Dessert (100g serving)</TableCell>
            <TableCell align="right">Power&nbsp;(Watts)</TableCell>
            <TableCell align="right">Automatic Programs</TableCell>
            <TableCell align="right">Manual Programs</TableCell>
            <TableCell align="right">Total Crushing Blades</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row) => (
            <TableRow key={row.name}>
              <TableCell component="th" scope="row">
                {row.name}
              </TableCell>
              <TableCell align="right">{row.power}</TableCell>
              <TableCell align="right">{row.autoPrograms}</TableCell>
              <TableCell align="right">{row.manPrograms}</TableCell>
              <TableCell align="right">{row.crushing}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
