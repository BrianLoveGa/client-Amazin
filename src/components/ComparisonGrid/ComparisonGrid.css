.tiny {
  display: none;
}

.productPic {
  width: 150px;
  height: 150px;
}
