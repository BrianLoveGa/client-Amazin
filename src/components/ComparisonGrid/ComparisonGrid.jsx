import React from 'react';
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import axios from 'axios';

class ComparisonGrid extends React.Component {  
  constructor(props) {  
    super(props)  
    this.state = {  
      items: [] 
    }  
  }  
  componentDidMount() {  
    axios.get("http://localhost:8080/all/products/").then(response => {   
      this.setState({  
        items: response.data  
      });  
    });  
  }  

  render() {  
    console.log(this.state.items);
    return (  
      <div>
       <TableContainer component={Paper}>  
        <Table stickyHeader  aria-label="sticky table">  
          <TableHead>  
            <TableRow> 
              <TableCell>Category</TableCell>
              {
                this.state.items.map(item => 
                  <TableCell align="right">{item.itemName}</TableCell>
                )
              }
            </TableRow> 
            </TableHead>  
          <TableBody>  
            {  
              this.state.items.map(item => {  
                return <TableRow>  
                  <TableCell component="th" scope="row">  
                    {item.itemId}  
                  </TableCell>   
                  <TableCell align="right">{item.itemAbout}</TableCell>   
                  <TableCell align="right">{item.itemPrice}</TableCell>    
                </TableRow>  
              })  
            }  
          </TableBody>  
        </Table>  
        </TableContainer>   
      </div>
    );  
  }  
}  
  
export default ComparisonGrid;