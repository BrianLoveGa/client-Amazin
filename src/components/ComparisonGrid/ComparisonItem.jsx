import React, { Component } from 'react';

export default function ComparisonItem() {
  return ""
  // <h1>Hello </h1>;
}