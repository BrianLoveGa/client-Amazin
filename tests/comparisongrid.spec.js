import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonGrid from '../src/components/ComparisonGrid/ComparisonGrid.jsx';

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ComparisonGrid/>, div);
})