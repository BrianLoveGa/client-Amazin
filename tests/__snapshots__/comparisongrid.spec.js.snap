// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    Hello 
  </h1>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            data-testid="comparisonGrid"
            scope="col"
          >
            Category
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Product #1
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Product #2
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Product #3
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Product #4
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Power
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            1100
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            1000
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            1200
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            1300
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Automatic Modes
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            0
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            4
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            4
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            0
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Manual Modes
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            4
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            0
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            4
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Price ($USD)
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            305
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            307
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            607
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            403
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Rating
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            2
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            3
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            5
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            3
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            Hello
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            2
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            3
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            5
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
